### PR TITLE
Ft/subscribe to post

### PR DESCRIPTION
**Pull request by rubygkendrick**
User is able to subscribe to an author on the post details page

added a subscribe button with handler function to PostDetails.jsx
added SubscriptionManager.js  with a POST request function 
added SubscriptionController with an endpoint to create a new subscription

To Test:
go to Post Details page and click Subscribe to Author. You should see a window alert confirming your subscription
Check PG Admin to make sure that the subscription has been successfully added with the appropriate IDs
 
Note: there is no functionality yet implemented to prevent a user from subscribing endlessly to another user. This could be done with a conditional rendering of the buttons for subscribe and unsubscribe on client side, or prevented via logic in the endpoint.

### CODE REVIEW

**Review by LucasBress0n — APPROVED** [review]

File: Controllers/SubscriptionController.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    public SubscriptionController(TabloidDbContext context)
+    {
+        _db = context;
+    }
+
+
+   [HttpPost]
+   [Authorize]
+   public IActionResult CreateSubscription(Subscription newSubscription)
```
Comment (Lincoln-WIlcox) [inline Controllers/SubscriptionController.cs]: Consider making a DTO so only the properties the user should be able to set are exposed (`SubscriberId` and `FollowerId`). `SubbedDate` and `UnsubbedDate` shouldn't be exposed to the user as they are set by the endpoint, but more important is the Id, as the endpoint will fail if the Id is set to a value where there already exists a subscription with that Id.

File: client/src/components/PostDetails.jsx [hunk 2]
```diff
@@ -17,6 +19,17 @@ export default function PostDetails() {
         getAndResetPost();
     }, []);
 
+    const handleSubscribeClick = () => {
+        const newSubscription = 
+        {
+            subscriberId: post.author?.id,
+            followerId: user.id
+        }
+        console.log(user)
```
Comment (Lincoln-WIlcox) [inline client/src/components/PostDetails.jsx]: If this console log was for testing purposes and you don't need it anymore, it'd be appreciated if it was removed.

**Review by Lincoln-WIlcox — CHANGES_REQUESTED** [review]

File: Controllers/SubscriptionController.cs (hunk 1 above)
Comment (rubygkendrick) [reply Controllers/SubscriptionController.cs]: I was under the impression that a DTO was only used and created to present data to the client. If I am looking to update the database... wouldn't I just post directly to Subscriptions? As I am not returning any information to the client?

Comment (Lincoln-WIlcox) [reply Controllers/SubscriptionController.cs]: DTOs can also be used to represent which data the user can give. [Here ](https://ardalis.com/web-api-dto-considerations/) is an article about it, with a section on using DTOs to receive data under the Create header.

**Review by Lincoln-WIlcox — APPROVED** [review]
sweet

**Pull request merged**